src/alire/alire-config-edit.adb

```diff
@@ -111,25 +114,58 @@ package body Alire.Config.Edit is
    --------------
 
    function Filepath (Lvl : Level) return Absolute_Path is
+
+      type Old_New is (Old, Current);
+
+      --------------
+      -- Location --
+      --------------
+
+      function Location (Which : Old_New) return Absolute_Path is
+         File : constant String :=
+                  (case Which is
+                      when Old     => "config.toml",
+                      when Current => Paths.Settings_File_Name);
+      begin
+         case Lvl is
+            when Global =>
+               return Alire.Config.Edit.Path / File;
+            when Local =>
+               declare
+                  Candidate : constant String :=
+                                Directories.Detect_Root_Path;
+               begin
+                  if Candidate /= "" then
+                     --  This file cannot have a .toml extension or the root
+                     --  detection will not work.
+                     return Candidate / "alire" / File;
+                  else
+                     Raise_Checked_Error
+                       ("Can only be used in an Alire directory");
+                  end if;
+               end;
+         end case;
+      end Location;
+
    begin
-      case Lvl is
-         when Global =>
-            return Alire.Config.Edit.Path / "config.toml";
-         when Local =>
-            declare
-               Candidate : constant String :=
-                 Directories.Detect_Root_Path;
-            begin
-               if Candidate /= "" then
-                  --  This file cannot have a .toml extension or the root
-                  --  detection will not work.
-                  return Candidate / "alire" / "config.toml";
-               else
-                  Raise_Checked_Error
-                    ("Can only be used in an Alire directory");
-               end if;
-            end;
-      end case;
+      --  Migrate file on the spot if necessary. This file is transparent to
+      --  users so we do not emit visible messages.
+
+      if Directories.Is_File (Location (Old)) and then
+        not Directories.Is_File (Location (Current))
+      then
+         Trace.Debug ("Migrating settings file: "
+                      & Location (Old) & " --> " & Location (Current));
+         Adirs.Copy_File (Source_Name => Location (Old),
+                          Target_Name => Location (Current));
+
+         --  Leave an explanation just in case
```

Comment:
Could we prepend a little comment in `config.toml`? I think I would prefer that as a user.

Reply:
Added